okhttp/src/main/java/com/squareup/okhttp/CertificatePinner.java

```diff
@@ -180,9 +215,23 @@ private static ByteString sha1(X509Certificate x509Certificate) {
     private final Map<String, List<ByteString>> hostnameToPins = new LinkedHashMap<>();
 
     /**
-     * Pins certificates for {@code hostname}. Each pin is a SHA-1 hash of a
-     * certificate's Subject Public Key Info, base64-encoded and prefixed with
-     * "sha1/".
+     * Pins certificates for {@code hostname}.
+     *
+     * @param hostname lower-case host name or wildcard pattern such as {@code *.example.com}.
+     *                 <p>Wildcard pattern rules:
```

Comment:
Wanna extract this out to a top-level comment, at the top of the class? This is interesting stuff and hiding it in the `@param` doc is buring a cool feature.

Also want to use `<ol><li>` instead of `<p>1.` ? That's what we do elsewhere.